### PR TITLE
Try to fix Travis master build for macOS 10.11

### PR DESCRIPTION
**Pull request by uklotzde**
https://builds.renegadetech.mixxx.org/job/master-release/955/architecture=amd64,platform=macosx

### CODE REVIEW

**Comment by uklotzde** [comment]
Whitelisting:
- Reverted back to the original, custom implementation for the general case
- Use aligned allocaction only if available and obey preconditions for parameters

File: src/util/sample.cpp [hunk 1]
```diff
@@ -53,8 +53,14 @@ CSAMPLE* SampleUtil::alloc(SINT size) {
         // boundaries.
         return static_cast<CSAMPLE*>(
                 _aligned_malloc(sizeof(CSAMPLE) * size, kAlignment));
-#elif defined(__GNUC__) && !defined(_GLIBCXX_HAVE_ALIGNED_ALLOC)
-        // On other platforms that don't support std::aligned_alloc
+#elif defined(_GLIBCXX_HAVE_ALIGNED_ALLOC)
+        std::size_t alloc_size = sizeof(CSAMPLE) * size;
+        // The size (in bytes) must be an integral multiple of kAlignment
+        std::size_t aligned_alloc_size = alloc_size + (kAlignment - alloc_size % kAlignment);
```
Comment (daschuer) [inline src/util/sample.cpp]: Even when the size is just right, it adds one kAlignment extra.
How about this
+= (kAlignment - 1) 
/= kAlignment
*= kAlignment

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/util/sample.cpp (hunk 1 above)
Comment (uklotzde) [reply src/util/sample.cpp]: Good catch :see_no_evil:

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Thank you. LGTM.

**Pull request merged**